chapter/semantics.tex

```diff
@@ -73,6 +84,28 @@ \section{Evaluation Contexts and Reduction Semantics}
   (v\;E)[e]  & = E[v e]
 \end{alignat*}
 
+When we associate evaluation contexts with stack frames, then these two variants
+correspond to two different orderings. In the inside-out style, the root contains
+the first operation that will be applied to the value produced by $\square$. \\
+
+Figures below show these contexts for $1 + (\square - 2)$.
+
+\begin{figure}[h]
+  \centering
+
+  \begin{subfigure}{0.30\textwidth}
+    \includegraphics[width=0.6\linewidth]{images/semantics/outside-in_context.jpg}
```

Comment:
I suggest to not include raster images into repository. It would be better to leave some placeholder here, or typeset such an image in e.g., tikz.

Reply:
Ideally, a tikz picture. But at this stage it can be some ascii-art, short description or even "image placeholder" text.